sdk/storage/azure_storage_blob/src/lib.rs

```diff
@@ -2,13 +2,34 @@
 //
 // Licensed under the MIT License. See License.txt in the project root for license information.
 // Code generated by Microsoft (R) Rust Code Generator. DO NOT EDIT.
+#![allow(dead_code)]
+#![allow(unused_imports)]
 
 pub mod clients;
 mod generated;
-pub use generated::*;
 mod pipeline;
 
-// Moved from src/generated/mod.rs to surface these specific Options bags to the crate root
-// pub use crate::generated::clients::{
-//     BlobClientOptions, BlobContainerClientOptions, BlobServiceClientOptions,
-// };
+pub use crate::clients::{BlobClient, BlobContainerClient, BlobServiceClient};
+pub use crate::generated::clients::{
+    BlobClientOptions, BlobContainerClientOptions, BlobServiceClientOptions,
+};
+pub use crate::generated::models::{
+    BlobClientDownloadOptions, BlobClientGetPropertiesOptions, BlobContainerClientCreateOptions,
+    BlobContainerClientDeleteOptions, BlobContainerClientGetPropertiesOptions,
+    BlobServiceClientGetPropertiesOptions, BlockBlobClientCommitBlockListOptions,
+    BlockBlobClientGetBlockListOptions, BlockBlobClientStageBlockOptions,
+    BlockBlobClientUploadOptions,
+};
+
+pub mod models {
+    pub use crate::generated::models::{
+        AccessTier, ArchiveStatus, BlobClientDownloadResult, BlobClientDownloadResultHeaders,
+        BlobClientGetPropertiesResult, BlobClientGetPropertiesResultHeaders,
+        BlobContainerClientCreateResult, BlobContainerClientDeleteResult,
+        BlobContainerClientGetPropertiesResult, BlobContainerClientGetPropertiesResultHeaders,
+        BlobImmutabilityPolicyMode, BlobType, BlockBlobClientCommitBlockListResult,
+        BlockBlobClientStageBlockResult, BlockBlobClientUploadResult, BlockList, BlockListType,
+        BlockLookupList, CopyStatus, LeaseState, LeaseStatus, PublicAccessType, RehydratePriority,
+        StorageServiceProperties,
+    };
```

Comment:
For now, I have lumped in the request models here, but we could look to move this to the root:
`BlockList`, `BlockListType`, `BlockLookupList`, and `StorageServiceProperties`